### PR TITLE
Update release_template.md

### PR DESCRIPTION
**Pull request by HanaGemela**

### CODE REVIEW

File: .github/release_template.md [hunk 1]
```diff
@@ -47,17 +44,14 @@ __TODO__:
 * Manual tests:
    * [ ] QA: [Changelog](https://github.com/owncloud/client/blob/master/CHANGELOG.md) test
    * [ ] QA: Regression test
-* [ ] QA: If required: create a separate test plan ticket for Windows VFS testing from [VFS Template](https://github.com/owncloud/QA/blob/master/Desktop/Test_Plan_VFS.md) - add the link here
```
Comment (fmoc) [inline .github/release_template.md:50]: Are these VFS tests no longer relevant to us? Should they be merged into this template?

CC @TheOneRing

**Comment by fmoc** [comment]
@michaelstingl please re-review

**Review by fmoc — APPROVED** [review]
Fine for me, let's see what @michaelstingl thinks.

**Review by michaelstingl — APPROVED** [review]
LGTM 👍

**Pull request merged**